### DISCUSSION TITLE
Onion skin feature - Out of pegs / Shift and trace

### DISCUSSION THREAD

**Discussion by mshoek** [keiwando/toonsquid#49] [Ideas]
TvPaint has this feature called "out of pegs", which means you can move the onion skin images closer together. Also, you can scale and rotate them.  It helps a lot with maintaining proportions, and it's really useful for inbetweening. In Toom Boom it's called "shift and trace" but it's basically the same thing. 
Are you planning to make a feature like this? It's really useful, and so far none of the apps I've tried had this.

**Comment by AQuickSecond** [comment]
I'd recommend this feature as well, it makes the workflow more consistent with traditional, expediting certain work.

**Comment by mshoek** [comment]
I’ve just realized, basically it’s already possible with a transform layer!

**Reply by DWrightAnimation** [reply]
How so?

**Reply by mshoek** [reply]
https://user-images.githubusercontent.com/118011378/230714645-44947c08-17c0-4dab-9568-26ee6969902e.mp4

I made this little video. I hope it helps! Just make sure that you "lock" the position of the frame you'd like to draw on.

**Reply by DWrightAnimation** [reply]
Well, that's one way to do it, but it'd be inconvenient to have to add in a transform layer and delete it every time you want to do the shift and trace for every drawing.

**Reply by mshoek** [reply]
You can keep it and adjust the position keyframes. But I agree! It's just a workaround, and it could be more simple.
Initially, I was happy with this option, after working with Callipeg. It also had a transform layer, but when it was applied, you couldn't draw anything on the animation layer affected by it.

**Reply by DWrightAnimation** [reply]
Yeah, sometimes I wonder if I even made the right decision investing in Callipeg. I initially only got it because it was the only animation app I knew of that had brush smoothing and multi-layer support, two things Procreate lacked in its animation features. ToonSquid has a lot of Callipeg's features, and so much more, and at only 20% of the price.

**Comment by DWrightAnimation** [comment]
I believe OpenToonz has this, too.
